### PR TITLE
PR #5: Welcome page + Preferences Setup

### LINKED ISSUE #103: Need a welcome page that introduces the user to our extension
## 🚀 Feature Summary

@KamilaKinel brought up a very good point in one of our meetings - most apps have a welcome page that allow you to set up your app to your needs!

I want to incorporate this feature into our app, where we have a welcome page and a page where the user can initially set their preferences.

## 📋 Description

Without initial preferences being set, we run into an issue of information overload. There would be far too many events showing on the app that the user is most likely not interested in. And, while a user can definitely choose not to have any preferences, this takes away the hurdle of a user accidentally not setting their preferences accordingly.

## ✅ Acceptance Criteria

List out what should be true when the feature is considered complete:

- [ X ] Must be aesthetically please
- [ X ] Must be intuitive to interact with
- [ X ] Must share a similar format with the preferences component on the Settings page

---

> Please tag the appropriate team members or leave a comment if this request should be prioritized or discussed further.

@SebastianYepez will implement this in his PR 5.

### PR DESCRIPTION
**Pull request by SebastianYepez**
# Description

We needed a welcome page to introduce users to our app, as well as allow them to set up their preferences for the first time. This PR closes #103.

I developed this "Welcome" workflow to be extremely thorough and I believe I implemented every edge case. I tried my best to act as a consumer of the product, so if there's anything I missed be sure to let me know.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Manual Testing
- [X] Automated Unit Testing

## Testing Coverage
- Updated Preferences.test.jsx
- Created PreferencesSetup.test.jsx

image

# UML
PR 5 UML - SY)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# How to test:
Go to chrome://extensions and "Remove" your RebelRemind instance
Then, rebuild your extension and "Load Unpacked" in chrome://extensions as we did the first time a while back

This will act as if you have just downloaded the extension. The welcome page should pop up. Please test thoroughly and ensure everything works as you'd expect!

What the PR looks like:
image
image
image
image
image

### CODE REVIEW

**Comment by KamilaKinel** [comment]
@SebastianYepez Just tested looks great! One small thing I think would be cool to add would be some sort of feedback to the user if they try making alterations to sport on the bottom but do not have sport checked above as preference. This can be implemented any way you think works best, some ex:

- Preference checkbox getting highlighted if edit tries to be made but preference isn't selected
- pop-up error prompting user to select preference to enable changes
- ...etc

**Review by KamilaKinel — CHANGES_REQUESTED** [review]
@SebastianYepez Just tested looks great! One small thing I think would be cool to add would be some sort of feedback to the user if they try making alterations to sport on the bottom but do not have sport checked above as preference. This can be implemented any way you think works best, some ex:

- Preference checkbox getting highlighted if edit tries to be made but preference isn't selected
- pop-up error prompting user to select preference to enable changes
- ...etc

File: rebel-extension/src/pages/PreferencesSetup.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                        Finish Setup
+                    </button>
+                </div>
+            </div>
+            :
+            <>
+                <div className="page-background">
+                    <div className="welcome-container">
+                        <h1>🛑 Before we get started...</h1>
+                        <h3>Please login with Google. This will help us confirm you are a UNLV student and be used for future integrations.</h3>
```
Comment (KamilaKinel) [inline rebel-extension/src/pages/PreferencesSetup.jsx:65]: Maybe specify sign in with google with your student account.

File: rebel-extension/src/components/CanvasTokenManager.jsx [hunk 2]
```diff
@@ -56,30 +56,30 @@ const CanvasTokenManager = () => {
...
 
                 {/* Toggle Visibility Button */}
                 <button
                     onClick={() => setShowToken(!showToken)}
-                    className="absolute right-2 top-2"
+                    className="canvasShowBT"
                 >
                     {showToken ? "🙈 Hide" : "👁 Show"}
-                </button>
+                </button>r
```
Comment (daltog1) [inline rebel-extension/src/components/CanvasTokenManager.jsx]:
```suggestion
                </button>
```
Please remove this r. It is adding an extra r in both places the CanvasTokenManager is visible. 
image

image

File: rebel-extension/src/components/Preferences.jsx [hunk 3]
```diff
@@ -184,211 +228,594 @@ const Preferences = () => {
...
+                        <div style={{
+                            marginTop: "1rem", opacity: preferences.canvasIntegration ? 1 : 0.5,
+                            pointerEvents: preferences.canvasIntegration ? "auto" : "none", textAlign: "left"
+                        }} className="canvas-container">
+                            <span style={{ textAlign: "left", fontWeight: "bold", fontSize: "1rem" }}>Canvas Personal Access Token</span>
+                            <p>Your Canvas Personal Access Token is stored locally on your browser and used to integrate our app with Canvas!
+                                We do not store this token, or any of your personal information externally.
+                            </p>
+                            <p style={{ fontWeight: "bold" }}>On Canvas, go to Account {">"} Settings {">"} Approved Integrations {">"}  New Access Token {">"}  Enter "Rebel Remind"</p>
+                            <p>Copy and paste your token here. Make sure to be wary of the time limit you set! Be sure to securely save it. If you log out of our app,
```
Comment (daltog1) [inline rebel-extension/src/components/Preferences.jsx]: Could we change this to
"Copy and paste your token here. Make sure to be wary of the expiration date you set! Be sure to securely save your it. If you log out of our app, it will delete your token!"
Expiration date better fits the wording on the actual Canvas page.

File: rebel-extension/src/components/Preferences.jsx [hunk 4]
```diff
@@ -184,211 +228,594 @@ const Preferences = () => {
...
+                    >
+                        <div style={{
+                            marginTop: "1rem", opacity: preferences.canvasIntegration ? 1 : 0.5,
+                            pointerEvents: preferences.canvasIntegration ? "auto" : "none", textAlign: "left"
+                        }} className="canvas-container">
+                            <span style={{ textAlign: "left", fontWeight: "bold", fontSize: "1rem" }}>Canvas Personal Access Token</span>
+                            <p>Your Canvas Personal Access Token is stored locally on your browser and used to integrate our app with Canvas!
+                                We do not store this token, or any of your personal information externally.
+                            </p>
+                            <p style={{ fontWeight: "bold" }}>On Canvas, go to Account {">"} Settings {">"} Approved Integrations {">"}  New Access Token {">"}  Enter "Rebel Remind"</p>
```
Comment (daltog1) [inline rebel-extension/src/components/Preferences.jsx]: Could we change "Enter "Rebel Remind"" to "Enter "Rebel Remind" for the purpose" to better fit the wording on the Canvas page?

**Review by daltog1 — CHANGES_REQUESTED** [review]
Overall Sebastian, this PR is looking great! It's important for users to have a welcome page that lets them set their preferences initially. Otherwise, some users would never even know to set them later on. I left a couple of suggestions within the code just for some small changes. Your UML diagram and test instructions look great. One more suggestion I have is to try to get the test coverage a little bit higher by adding a couple more tests that should be able to hit some of those chunks. That way, we know some of those edge cases work as expected. Also, don't forget to update the welcome page extension screenshot when we make visual changes so new users don't see an out of date screenshot. Keep up the great work!

File: rebel-extension/src/components/Preferences.jsx (hunk 4 above)
Comment (SebastianYepez) [reply rebel-extension/src/components/Preferences.jsx]: Fixed to 'Enter "Rebel Remind" as the Purpose'

File: rebel-extension/src/components/Preferences.jsx (hunk 3 above)
Comment (SebastianYepez) [reply rebel-extension/src/components/Preferences.jsx]: Done!

File: rebel-extension/src/components/CanvasTokenManager.jsx (hunk 2 above)
Comment (SebastianYepez) [reply rebel-extension/src/components/CanvasTokenManager.jsx]: Thank you for pointing this out! Fixed in latest commit.

**Review by daltog1 — APPROVED** [review]
Requested code changes have been made. Nice job on this PR Sebastian!

**Review by KamilaKinel — APPROVED** [review]
@SebastianYepez Amazing work! I love what you added and it works perfectly.  Approved.

File: rebel-extension/src/pages/PreferencesSetup.jsx (hunk 1 above)
Comment (daltog1) [reply rebel-extension/src/pages/PreferencesSetup.jsx:65]: Or "Please login with Rebelmail"
